### PR TITLE
Update commerce event mapping

### PR DESCRIPTION
**Pull request by SyedWasiHaider**
# Summary
The purpose of this change is to provide an update to the Amplitude commerce mapping. This also an attempt to bring the JS integration up to parity with the server side integration. I've also added an adhoc test for ease of testing. The other notable change is that I'm using rawgit for the mparticle.js file (vs trying to recreate it entirely in the mock).

## Master Issue
https://github.com/mParticle/mPServer/issues/4538

## Test Plan
-See updated README.MD

### CODE REVIEW

**Comment by SyedWasiHaider** [comment]
cc @brianczako for awareness (as we just did something for Amplitude)

File: test/adhoc/index.html [hunk 1]
```diff
@@ -0,0 +1,180 @@
+<!DOCTYPE html>
```
Comment (rmi22186) [inline test/adhoc/index.html]: I wasn't able to get this working using python -m SimpleHTTPServer 8000 for some reason. amplitude won't load.

I am able to test the revenue functions just fine though on my own test app that. I made. my goal is to get a more polished test app that will be at least somewhat more user-friendly than the one I currently have :)

so I think we can leave this out of the commit, but if you're free, would like to chat about just getting this working if possible

Comment (rmi22186) [inline test/adhoc/index.html]: I'd leave this test file out. I'm going to try to get an environment up for people to be able to test things on. Right now I have a pretty hacky way to do all the live testing but it' like to polish this process more.

Also - I wasn't able to get this going even with just a simple python server ( python -m SimpleHTTPServer 8000. It wouldn't load amplitude, but the manual testing worked fine using my test app. 

Would be great to hop on a quick share to see what I'm doing wrong with this file though just as an FYI to me :)

File: test/index.html [hunk 2]
```diff
@@ -102,12 +104,23 @@
         };
 
         window.amplitude = new amp();
-        window.mParticle = new mp();
+        window.amplitude.Revenue = function(){
```
Comment (rmi22186) [inline test/index.html]: small nit - maybe move this into the amp constructor above?

File: Amplitude.js [hunk 3]
```diff
@@ -180,17 +180,32 @@
...
+
+                var isRefund = event.ProductAction.ProductActionType == mParticle.ProductActionType.Refund;
+                var logRevenue = (event.ProductAction.ProductActionType == mParticle.ProductActionType.Purchase) || isRefund;
+                var expandedEvents = mParticle.eCommerce.expandCommerceEvent(event);
+                
+                expandedEvents.forEach(function(expandedEvt) {
+
+                    // Exclude Totals from the attributes as we log it in the revenue call
+                    var updatedAttributes = {};
+                    Object.keys(expandedEvt.EventAttributes).filter(function (key) { return key != "Total Amount" && key != "Total Product Amount" } ).forEach(function(key) {
```
Comment (SyedWasiHaider) [inline Amplitude.js]: I think it should be 

    if (key !== 'Total Amount' && key !== 'Total Product Amount') {

I'm guessing just a typo

**Comment by SyedWasiHaider** [comment]
Oh strange. I forgot to actually commit my README change.

**Comment by SyedWasiHaider** [comment]
@rmi22186 Thanks for the feedback.

I've made most of the changes you asked for.

With regards to the adhoc test file, make sure you run the python server in the root of the folder. If it still doesn't work, I can delete it.

**Comment by rmi22186** [comment]
LGTM - got it working. i was trying to python simple server from the adhoc folder

**Comment by samdozor** [comment]
@SyedWasiHaider can you make sure you either squash or clean up the commits before merging? Thanks :)

**Pull request merged**